tests/run_tests.py

```diff
@@ -303,32 +316,34 @@ def build_jit_dependencies() -> None:
 
 
 class Arguments(Tap):
-    test: Path | None = None
+    test: list[Path] = [TESTS_DIR]
     workers: int = cpu_count()
 
+    def configure(self) -> None:
+        self.add_argument("test", nargs="*")
+
 
 def main() -> None:
     args = Arguments().parse_args()
 
-    assert TESTS_DIR.is_dir()
     OUT_DIR.mkdir(exist_ok=True)
 
     build_jit_dependencies()
 
-    if args.test is not None:
-        test_path = args.test if args.test.is_file() else PARENT_DIR / args.test
-        try:
-            run_test_print_result(test_path.resolve(strict=True))
-        except FileNotFoundError:
-            print(f"No such file: '{test_path}'")
-    else:
-        all_test_files = [
-            test_file
-            for test_file in TESTS_DIR.rglob("**/*.c3")
-            if not test_file.stem.startswith("_") and test_file.is_file()
-        ]
+    all_test_files: set[Path] = set()
+    for file_or_dir in args.test:
+        if file_or_dir.is_file():
+            all_test_files.add(file_or_dir.resolve(strict=True))
+        elif file_or_dir.is_dir():
+            all_test_files.update(
+                test_file.resolve(strict=True)
+                for test_file in file_or_dir.rglob("**/*.c3")
+                if not test_file.stem.startswith("_") and test_file.is_file()
+            )
+        else:
+            print(f"ERROR: Invalid path {file_or_dir}")
 
-        sys_exit(run_tests(all_test_files, args.workers))
+    sys_exit(run_tests(sorted(all_test_files), args.workers))
```

Comment:
While this is objectively an improvement I did like it when they were sorted chronologically and my new test was at the bottom.

Reply:
I'm not sure it's worth replicating that and I don't want tests to run in a non-deterministic order :sweat_smile: